Makefile

```diff
@@ -90,6 +101,28 @@ test-silent:
 benchmark-test:
 	. ./scripts/shared_env && go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
 
+test-artifacts: out
+	go test -race -tags integration -o ./out/test-artifacts/unix-engine-tests -c ./agent/engine
```

Comment:
I think it'd be nicer to split these into targets that control the files, or at least have the generated files in the target.  Kinda like this:

```make
SOURCEDIR=./agent

.PHONY: test-artifacts
test-artifacts: linux-integration-artifacts linux-functional-artifacts

.PHONY: linux-integration-artifacts
linux-integration-artifacts: out/test-artifacts/linux-stats-tests out/test-artifacts/unix-app-tests

INTEG_STATS_SOURCES := $(shell find $(SOURCEDIR)/stats -name '*.go')
INTEG_APP_SOURCES := $(shell find $(SOURCEDIR)/app -name '*.go')
out/test-artifacts/linux-stats-tests out/test-artifacts/linux-app-tests: $(INTEG_STATS_SOURCES) $(INTEG_APP_SOURCES)
	go test -race -tags integration -o ./out/test-artifacts/linux-stats-tests -c ./agent/stats
	go test -race -tags integration -o ./out/test-artifacts/linux-app-tests -c ./agent/app
```

Reply:
neat